docs/configs-npb-tests/run_npb.py

```diff
@@ -93,42 +99,50 @@ def writeBenchScript(dir, bench):
 
     globalStart = time.time()
 
-
     print("Running the simulation")
+    print("Using cpu: {}".format(cpu))
     exit_event = m5.simulate()
 
-    # While there is still something to do in the guest keep executing.
-    # This is needed since we exit for the ROI begin/end
-    foundROI = False
-    while exit_event.getCause() != "m5_exit instruction encountered":
-        # If the user pressed ctrl-c on the host, then we really should exit
-        if exit_event.getCause() == "user interrupt received":
-            print("User interrupt. Exiting")
-            break
+    if exit_event.getCause() == "workbegin":
+        # Reached the start of ROI
+        # start of ROI is marked by an
+        # m5_work_begin() call
+        print("Resetting stats at the start of ROI!")
+        m5.stats.reset()
+        start_tick = m5.curTick()
+        start_insts = system.totalInsts()
+        # switching to atomic cpu if argument cpu == atomic
+        if cpu == 'atomic':
+            system.switchCpus(system.cpu, system.atomicCpu)
+    else:
+        print("Unexpected termination of simulation !")
```

Comment:
It would be a good idea to print why the simulation exited here. It will make debugging easier.